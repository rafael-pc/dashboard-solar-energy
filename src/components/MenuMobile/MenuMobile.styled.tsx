import styled from "styled-components";
import { theme } from "../../styles";

export const StyledMenu = styled.nav<{ open: boolean }>`
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  background: ${theme.colors.bg_primary};
  transform: ${({ open }) => (open ? "translateX(0)" : "translateX(-100%)")};
  height: 100vh;
  text-align: left;
  padding: 2rem;
  position: absolute;
  transition: transform 0.3s ease-in-out;

  @media (max-width: ${theme.screen.mobile}) {
    width: 100%;
  }

  a {
    text-transform: uppercase;
    padding: 2rem 0;
    font-weight: ${theme.font.bold};
    letter-spacing: 0.5rem;
    color: ${theme.colors.neutral_100};
    text-decoration: none;
    line-height: 1.5em;
    transition: color 0.3s linear;
    font-size: ${theme.font.sizes.xs};
    text-align: center;

    &:hover {
      color: ${theme.colors.neutral_300};
    }
  }
`;
