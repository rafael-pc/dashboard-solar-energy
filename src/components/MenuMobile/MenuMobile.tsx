import { FC } from "react";
import { StyledMenu } from "./MenuMobile.styled";
import { Link } from "react-router-dom";

interface IMenuMobileProps {
  isOpen: boolean;
  id: string;
}

const MenuMobile: FC<IMenuMobileProps> = ({ isOpen, id, ...props }) => {
  const isHidden = isOpen ? true : false;
  const tabIndex = isHidden ? 0 : -1;

  return (
    <StyledMenu open={isOpen} id={id} aria-hidden={!isHidden} {...props}>
      <Link to="/dashboard" tabIndex={tabIndex}>
        Dashboard
      </Link>
      <Link to="/units" tabIndex={tabIndex}>
        Unidade consumidora
      </Link>
      <Link to="/register_month" tabIndex={tabIndex}>
        Cadastro de en√©rgia gerada
      </Link>
    </StyledMenu>
  );
};

export default MenuMobile;
