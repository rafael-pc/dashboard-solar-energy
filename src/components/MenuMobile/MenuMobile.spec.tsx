import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MenuMobile from './MenuMobile';
import "@testing-library/jest-dom";

describe('MenuMobile', () => {
  it('renders links when open', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <MenuMobile isOpen={true} id="menu-mobile" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByText('Dashboard')).toBeInTheDocument();
    expect(screen.getByText('Unidade consumidora')).toBeInTheDocument();
    expect(screen.getByText('Cadastro de en√©rgia gerada')).toBeInTheDocument();
  });
});
