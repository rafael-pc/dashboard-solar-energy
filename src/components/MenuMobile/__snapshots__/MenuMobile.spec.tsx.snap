// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMobile renders links when open 1`] = `
<DocumentFragment>
  <nav
    aria-hidden="false"
    class="sc-bcPKhP prLgQ"
    id="menu-mobile"
    open=""
  >
    <a
      href="/dashboard"
      tabindex="0"
    >
      Dashboard
    </a>
    <a
      href="/units"
      tabindex="0"
    >
      Unidade consumidora
    </a>
    <a
      href="/register_month"
      tabindex="0"
    >
      Cadastro de en√©rgia gerada
    </a>
  </nav>
</DocumentFragment>
`;
